### PR TITLE
add tooltip & cell-box image

### PR DESCRIPTION
**Pull request by yfeng2824**
**Description:**
1. Create a Tooltip component for users to quickly view the explanation of terms without navigating away. 
2. Replace some internal links with this Tooltip component
Screenshot 2024-06-07 at 6 20 34 PM

3. Tooltip contents are auto-generated by generate-keyTerms.js (we can run yarn gen-term to re-generate the JSON file)
4. Add the cell-as-box image to how-ckb-works

---

**Guide for Using the Tooltip Component in MDX Files**
Step 1: Import the Tooltip Component
`import Tooltip from '@components/Tooltip';`

Step 2: Use the Tooltip Component
Simply wrap the term you want to highlight with <Tooltip>. For example:
`<Tooltip>UTXO</Tooltip>`

Step 3: Apply Tooltip to the First Occurrence
Only apply the tooltip to the first occurrence of the term in the file that is not explained in the following paragraphs. This will ensure clarity and prevent redundancy.

### CODE REVIEW

File: website/src/components/Tooltip/keywordList.ts [hunk 1]
```diff
@@ -0,0 +1,69 @@
+export interface GlossaryTerm {
+  link?: string;
+  content: string;
+}
+
+export const keywordList: Record<string, GlossaryTerm> = {
```
Comment (RetricSu) [inline website/src/components/Tooltip/keywordList.ts:6]: I feel like it is a burden to maintain two files for the same term explaination. How about we create a  `website/scripts/generate-term.js`, and a command `'gen-term: node scripts/generate-term.js'`, we then run  `yarn gen-term` the script reads from the glossary.mdx and write each term a `term.json` file. the keywordList can read from this `term.json`? in that way, we only maintains the glossary.mdx file.

File: website/src/components/Tooltip/index.tsx [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+const Tooltip: React.FC<TooltipProps> = ({ className, children }) => {
+  const term = children?.toString();
+  const normalizedTerm = term?.toLowerCase().replace(/\s+/g, "-");
+  const [tooltipContent, setTooltipContent] = useState<string>("");
+  // Default link points to the Glossary
+  const [tooltipLink, setTooltipLink] = useState<string>(
+    `/docs/tech-explanation/glossary#${normalizedTerm}`
+  );
+
+  useEffect(() => {
```
Comment (RetricSu) [inline website/src/components/Tooltip/index.tsx:21]: a little bit concern about `useEffect` here. can you check the academy implementation and see if it has better way to do this for performance? ref: https://react.dev/learn/you-might-not-need-an-effect

**Review by linnnsss — COMMENTED** [review]
👍

File: website/scripts/generate-keyTerms.mjs [hunk 3]
```diff
@@ -0,0 +1,82 @@
+import fs from "fs";
```
Comment (RetricSu) [inline website/scripts/generate-keyTerms.mjs:1]: please rename file to `generate-key-terms.mjs` for consistance

File: website/src/components/Tooltip/keyTerms.json
Comment (RetricSu) [inline website/src/components/Tooltip/keyTerms.json:1]: please rename file to `key-terms.json` for consistance

File: website/scripts/generate-keyTerms.mjs
Comment (RetricSu) [inline website/scripts/generate-keyTerms.mjs:1]: please add a command in `package.json` to run this files, and record the maintaince flow for key terms under https://github.com/nervosnetwork/docs.nervos.org?tab=readme-ov-file#develop

**Review by RetricSu — APPROVED** [review]

**Pull request merged**